testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6703,11 +6704,53 @@ private static void checkIf(SqlOperatorFixture f) {
     // test operands not in same type family.
     f.checkFails("^map_concat(map[1, null], array[1])^",
         "Parameters must be of the same type", false);
+
+    // 2. check with map function, map(k, v ...)
+    final SqlOperatorFixture f1 = fixture()
+        .setFor(SqlLibraryOperators.MAP_CONCAT)
+        .withLibrary(SqlLibrary.SPARK);
+    f1.checkScalar("map_concat(map('foo', 1), map('bar', 2))", "{foo=1, bar=2}",
+        "(CHAR(3) NOT NULL, INTEGER NOT NULL) MAP NOT NULL");
+    f1.checkScalar("map_concat(map('foo', 1), map('bar', 2), map('foo', 2))", "{foo=2, bar=2}",
+        "(CHAR(3) NOT NULL, INTEGER NOT NULL) MAP NOT NULL");
+    f1.checkScalar("map_concat(map(null, 1), map(null, 2))", "{null=2}",
+        "(NULL, INTEGER NOT NULL) MAP NOT NULL");
+    f1.checkScalar("map_concat(map(1, 2), map(1, null))", "{1=null}",
+        "(INTEGER NOT NULL, INTEGER) MAP NOT NULL");
+    // test zero arg, but it should return empty map.
+    f1.checkScalar("map_concat()", "{}",
+        "(VARCHAR NOT NULL, VARCHAR NOT NULL) MAP");
+
+    // test concat with empty map, in fact, spark will return MAP(CHAR, INTEGER), because
+    // it will add a cast 'cast(map() as map<string, int>)' to convert UNKNOWN type,
+    // but currently calcite not support cast to map type.
+    f1.checkScalar("map_concat(map('foo', 1), map())", "{foo=1}",
+        "(UNKNOWN NOT NULL, UNKNOWN NOT NULL) MAP NOT NULL");
+
+    // after calcite supports cast(null as map<string, int>), cast(map() as map<string, int>)
+    // it should add these tests.
+    if (TODO) {
```

Comment:
instead of `if (TODO)` would it be better to add an enum to the `Bug` class? Here is an example of what I mean: https://github.com/apache/calcite/blob/main/testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java#L624

Reply:
I get your point. but the form such as `cast(map() as map<string, int>)` may not necessarily be a bug, and there is currently no JIRA to track it, calcite is not ready to support it yet, and may not support it, so it is kept as a TODO for the time being.